pkg/startup/startup.go

```diff
@@ -1365,3 +1370,54 @@ func extractKubeadmCIDRs(kubeadmConfig *v1.ConfigMap) (string, string, error) {
 
 	return v4, v6, err
 }
+
+// Check if the string is cidr
+// Since: internal/bytealg/indexbyte_generic.go
+func indexByteString(s string, c byte) int {
+	for i := 0; i < len(s); i++ {
+		if s[i] == c {
+			return i
+		}
+	}
+	return -1
+}
+
+// return the cidr of the dest on local
+func getLocalCidr(ip string, version int) (string, error) {
+	i := indexByteString(ip, '/')
+	if i >= 0 {
+		log.Infof("%s is already cidr", ip)
+		return ip, nil
+	}
+	log.Infof("Auto-detecting IPv%d CIDR of %s", version, ip)
+
+	// Get a full list of interface and IPs and find the CIDR matching the
+	// found IP.
+	ifaces, err := autodetection.GetInterfaces(nil, nil, version)
+	if err != nil {
+		return ip, err
+	}
+	for _, iface := range ifaces {
+		log.WithField("Name", iface.Name).Info("Checking interface CIDRs")
+		for _, cidr := range iface.Cidrs {
+			log.WithField("CIDR", cidr.String()).Info("Checking CIDR")
+			var destCidr net.IP
```

Comment:
I think we can parse the IP address once, outside of the loop, rather than do it every time again inside the loop.